src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -604,6 +608,34 @@ OverlayImpl::size()
     return m_publicKeyMap.size ();
 }
 
+Json::Value
+OverlayImpl::crawl()
+{
+    Json::Value jv;
+    auto& av = jv["active"] = Json::Value(Json::arrayValue);
+    std::lock_guard <decltype(mutex_)> lock (mutex_);
+    for (auto const& e : m_publicKeyMap)
+    {
+        auto const sp = e.second.lock();
+        if (sp)
+        {
+            auto& pv = av.append(Json::Value(Json::objectValue));
+            pv["type"] = "peer";
+            pv["public_key"] = beast::base64_encode(
+                sp->getNodePublic().getNodePublic().data(),
+                    sp->getNodePublic().getNodePublic().size());
+            if (sp->crawl())
+            {
+                if (sp->slot()->inbound())
+                    pv["ip"] = sp->getRemoteAddress().address().to_string();
```

Comment:
Works for me.

On Thu, Jan 29, 2015 at 3:37 PM, Vinnie Falco <notifications@github.com>
wrote:

> In src/ripple/overlay/impl/OverlayImpl.cpp
> <https://github.com/ripple/rippled/pull/839#discussion_r23800432>:
>
> > +    auto& av = jv["active"] = Json::Value(Json::arrayValue);
> > +    std::lock_guard <decltype(mutex_)> lock (mutex_);
> > +    for (auto const& e : m_publicKeyMap)
> > +    {
> > +        auto const sp = e.second.lock();
> > +        if (sp)
> > +        {
> > +            auto& pv = av.append(Json::Value(Json::objectValue));
> > +            pv["type"] = "peer";
> > +            pv["public_key"] = beast::base64_encode(
> > +                sp->getNodePublic().getNodePublic().data(),
> > +                    sp->getNodePublic().getNodePublic().size());
> > +            if (sp->crawl())
> > +            {
> > +                if (sp->slot()->inbound())
> > +                    pv["ip"] = sp->getRemoteAddress().address().to_string();
>
> We don't want that, because it doesn't reflect the connectivity of the
> network. If other peers have outgoing connections to that peer, then we
> will discover it during the crawl.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ripple/rippled/pull/839/files#r23800432>.
>



-- 
*Edward Hennis | *Senior Software Engineer | *Ripple Labs Inc.*
ed@ripple.com | www.ripple.com | *www.ripplelabs.com
<http://www.ripplelabs.com/>*
T. 352.327.8769